src/config.rs

```diff
@@ -155,3 +158,50 @@ create_config! {
     cfg_test: bool, true, false, "build cfg(test) code";
     unstable_features: bool, false, false, "enable unstable features";
 }
+
+/// A rustfmt config (typically specified via rustfmt.toml)
+pub struct FmtConfig(rustfmt::config::Config);
+
+impl FmtConfig {
+    /// Look for `.rustmt.toml` or `rustfmt.toml` in `path`, falling back
+    /// to the default config if neither exist
+    pub fn from(path: &Path) -> FmtConfig {
+        const FMT_CONFIG_NAMES: [&str; 2] = [".rustfmt.toml", "rustfmt.toml"];
+        for fmt_config_name in &FMT_CONFIG_NAMES {
+            let config_path = path.to_owned().join(fmt_config_name);
+            let config_file = File::open(config_path);
+            if let Ok(mut f) = config_file {
+                let mut toml = String::new();
+                f.read_to_string(&mut toml).unwrap();
+                if let Ok(config) = rustfmt::config::Config::from_toml(&toml) {
+                    let mut config = FmtConfig(config);
+                    config.set_rls_options();
+                    return config;
+                }
+            }
+        }
+        FmtConfig::default()
+    }
+
+    // options that are always used when formatting with rls
+    fn set_rls_options(&mut self) {
+        self.0.set().skip_children(true);
+        self.0.set().write_mode(WriteMode::Plain);
+    }
+}
+
+impl Default for FmtConfig {
+    fn default() -> FmtConfig {
+        let config = rustfmt::config::Config::default();
+        let mut config = FmtConfig(config);
+        config.set_rls_options();
+        config
+    }
+}
+
+impl Deref for FmtConfig {
```

Comment:
I don't like this pattern of using `Deref` for newtypes, see https://github.com/rust-unofficial/patterns/blob/master/anti_patterns/deref.md for why